lib/buffer.js

```diff
@@ -635,8 +636,8 @@ Buffer.prototype.toString = function(encoding, start, end) {
 
 Buffer.prototype.equals = function equals(b) {
   if (!isUint8Array(b))
-    throw new TypeError('Argument must be a Buffer or Uint8Array');
-
+    throw new errors.TypeError('ERR_INVALID_ARG_TYPE', 'argument',
```

Comment:
same comment here as before, probably need to check for all cases that the second string matches the name of the argument that was invalid.